telegram_bot/main.py

```
@@ -0,0 +1,150 @@
+
+
+def _apply_access_control(app, allowed_user_ids: set[int]) -> None:
+    """
+    Add a pre-handler that rejects updates from users not in the allow-list.
+
+    Uses a TypeHandler with group=-1 so it runs before any command handler.
+    """
+    from telegram import Update
+    from telegram.ext import TypeHandler
+
+    async def _check_user(update: Update, context) -> None:
```

Ruff (0.15.4)

[warning] Unused function argument: `context`

(ARG001)